firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionImpl.java

```diff
@@ -437,14 +450,18 @@ private void doStartFeedback(CharSequence infoText, @Nullable Uri screenshotUri)
   }
 
   private Task<Void> launchFeedbackActivity(
-      @Nullable String releaseName, CharSequence infoText, @Nullable Uri screenshotUri) {
+      @Nullable String releaseName,
+      CharSequence infoText,
+      @Nullable Uri screenshotUri,
+      FeedbackTrigger trigger) {
     return lifecycleNotifier.consumeForegroundActivity(
         activity -> {
           LogWrapper.i(TAG, "Launching feedback activity");
           Intent intent = new Intent(activity, FeedbackActivity.class);
           // in development-mode the releaseName might be null
           intent.putExtra(FeedbackActivity.RELEASE_NAME_KEY, releaseName);
           intent.putExtra(FeedbackActivity.INFO_TEXT_KEY, infoText);
+          intent.putExtra(FeedbackActivity.FEEDBACK_TRIGGER_KEY, trigger.toString());
```

Comment:
You probably should use `name()` instead of `toString()` here if you keep `valueOf()` for deserialization.

Also, maybe there's a simple way to test that this parameter is passed (i.e. serialized and deserialized) correctly.

Reply:
Created a `fromString()` method for deserialization. It would be nice to test this, but we don't have any tests for the Activity where the deserialization is performed. I did test this locally and it seems to work. I'm gonna time-box an investigation into adding an test for FeedbackActivity.

Reply:
Activity testing is hard. Going to punt on this for now but may revisit it this week.